### DISCUSSION TITLE
Extremely long runtime for TinyVGG in chapter 4

### DISCUSSION THREAD

**Discussion by Romsha** [mrdbourke/pytorch-deep-learning#912] [Q&A]
Hello all,
I am now following the forth chapter and I'n getting very long train/test runtimes for model_0 (TinyVGG).
The example in the book ran in 4.935 seconds (for 5 epochs), yet on my computer each epoch takes 40-70 seconds!
I know there's a difference in the hardware (I'm running the code on my Mac M1 and not Colab+GPU) but up until now there wasn't such a difference in the runtimes.
Did anybody else encountered this issue?
[I pulled the notebook from this repo and ran it as-is, so this can't be a typo/coding error on my side]

P.S
I tried timing the test step in more detail, and it seems that most of the runtime is coming from the loop retrieving the values from the dataloader iterator. In addition, running the model on a single example image is very fast, so I really don't think this is a computational issue (at least not from the model itself).

**Comment by LuluW8071** [comment]
Maybe you didn't setup device agnostic code for M1 chip GPU.
Refer to this #337

**Comment by bofcarbon1** [comment]
I have chosen to only use CPU and work in PyCharm. On a CPU only it does take a while to run. 
Granted I cannot measure the optimization speed of the GPU this way. 
I will also add that using 10 hidden units resulted in 7 of 9 correct predictions of images.
Changing the hidden units to 30 resulted in 8 of 9 correct predictions.
Finally...
Changing the hidden units to 60 resulted in 9 of 9 correct predictions. 
Which took  considerably more time. Yes. Did not record  an exact time.

```
model_2 = FashionMNISTModelV2(input_shape=1,
                                  hidden_units=60,
                                  output_shape=len(class_names)).to(device)

```
